### PR TITLE
Fetch error handling refactor

### LINKED ISSUE #3815: Error HOC doesnt ever catch handle errors
**Is your feature request related to a problem? Please describe.**

We have a [`WithError` HOC](https://github.com/bbc/simorgh/blob/MpulsePageTypeService/src/app/containers/PageHandlers/withError/index.jsx) which shows an error page if `error` is truthy. Our issue is, thats basically never the case.

Error is set [here](https://github.com/bbc/simorgh/blob/MpulsePageTypeService/src/app/containers/App/App.jsx#L55-L69). If `route.getInitialData` throws an error, we catch it and set `error`. This is great, except our data fetching [already has a try catch inside it](https://github.com/bbc/simorgh/blob/MpulsePageTypeService/src/app/routes/getInitialData/utils/fetchData/index.js#L12-L29) meaning the error pretty much never propagates out.

This means all error pages caused by bad data get caught by the [withData](https://github.com/bbc/simorgh/blob/MpulsePageTypeService/src/app/containers/PageHandlers/withData/index.jsx#L51) HOC instead.

**Describe the solution you'd like**
Simply our error handling, and ideally expand it. We should only have 1 error page visual route.

**Describe alternatives you've considered**

**Testing notes**
This is mainly an investigation issue for now.

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [ ] This feature is expected to need manual testing.

**Additional context**

### PR DESCRIPTION
**Pull request by jroebu14**
Resolves https://github.com/bbc/simorgh/issues/3815

**Overall change:** Fixes error page not being shown in most cases.

Removes error handling from `fetchData` call in `App.jsx` and does error handling inside of `fetchData` which now also returns an error object that we can pass to the error HOC to detect any errors and show the relevant error page to the user.

**Code changes:**

- Removes error handling from `fetchData` call in `App.jsx`
- Refactors `fetchData` to make it easier to reason about
- `fetchData` now returns error which can be passed to error HOC

**Testing notes**
It's possible to simulate 500 and 404 errors by changing the code in `src/app/routes/getInitialData/utils/fetchData/index.js`:

404 - change the line with `fetch(url)` to `fetch(url + 'blah')`
500 - after the `handleResponse` function definition (line 11) add `throw new Error();` to the next line (line 12)

By doing this you should see the relevant error pages.

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [x] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`) 
- [x] This PR requires manual testing

### CODE REVIEW

File: src/app/containers/App/App.jsx [hunk 1]
```diff
@@ -55,28 +56,15 @@ export const App = ({ routes, location, initialData, bbcOrigin, history }) => {
...
-        }
-      };
-      fetchData();
+      route.getInitialData(match.params).then(data =>
+        setState(prevState => ({
+          ...prevState,
+          loading: false,
+          pageData: path(['pageData'], data),
+          status: path(['status'], data),
+          error: path(['error'], data),
```
Comment (dr3) [inline src/app/containers/App/App.jsx:65]: We need to handle `error` on line 28 too. As it stands, errors will be handled on onward navigation differently to the initial server render.

**Review by dr3 — DISMISSED** [review]

File: src/app/containers/App/App.jsx (hunk 1 above)
Comment (jroebu14) [reply src/app/containers/App/App.jsx:65]: @dr3 @mykeels fixed by destructuring `error` from `initialData` and using it in the `initialState`.

**Review by ryanmccombe — APPROVED** [review]

**Review by EinsteinNjoroge — APPROVED** [review]
LGTM

**Review by oluoluoxenfree — APPROVED** [review]

**Review by dr3 — APPROVED** [review]

**Comment by PriyaKR** [comment]
Looks good to me.

**Review by thekp — APPROVED** [review]

**Pull request closed without merging**